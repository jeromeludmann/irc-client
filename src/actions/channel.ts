import { ChannelScope } from "@app/types";

export const ChannelListTypes = {
  ADD: "CHANNEL/ADD",
  REMOVE: "CHANNEL/REMOVE",
};

export interface AddChannelAction {
  type: typeof ChannelListTypes.ADD;
  payload: ChannelScope;
}

export interface RemoveChannelAction {
  type: typeof ChannelListTypes.REMOVE;
  payload: ChannelScope;
}
