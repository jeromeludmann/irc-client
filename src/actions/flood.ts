export const SET_FLOOD_ON = 'FLOOD/ON'
export const SET_FLOOD_OFF = 'FLOOD/OFF'

export function setFlood(active: boolean) {
  return {
    type: active ? SET_FLOOD_ON : SET_FLOOD_OFF,
  }
}
