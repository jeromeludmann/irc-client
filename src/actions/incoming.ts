import { Action } from "redux";
import { Route, STATUS, BROADCAST } from "@app/Route";

export interface IncomingMessageAction<T, M> extends Action<T> {
  payload: M;
  route: Route;
}

export type IncomingMessageActionCreator<A, P = void> = (
  serverKey: string,
  prefix: P,
  params: string[],
) => A;

// TODO
// export interface Server {
//   name: string;
// }
export type Server = string;

export interface User {
  nick: string;
  user: string;
  host: string;
}

export type Prefix = undefined | Server | User;

export const isChannel = (channel: string) => /^(&|#|\+|!)/.test(channel);

export const isPrefixServer = (prefix: Prefix) => typeof prefix === "string";

export const isPrefixUser = (prefix: Prefix) => !isPrefixServer(prefix);

// ERROR

interface Error {
  message: string;
}

export const ERROR = "MESSAGE/INCOMING/ERROR";

export type ErrorAction = IncomingMessageAction<typeof ERROR, Error>;

export const receiveError: IncomingMessageActionCreator<ErrorAction> = (
  serverKey,
  _,
  params,
) => ({
  type: ERROR,
  payload: { message: params[0] },
  route: { serverKey, channelKey: BROADCAST },
});

// JOIN

interface Join {
  user: User;
  channel: string;
}

export const JOIN = "MESSAGE/INCOMING/JOIN";

export type JoinAction = IncomingMessageAction<typeof JOIN, Join>;

export const receiveJoin: IncomingMessageActionCreator<JoinAction, User> = (
  serverKey,
  user,
  params,
) => ({
  type: JOIN,
  payload: { user, channel: params[0] },
  route: { serverKey, channelKey: params[0] },
});

// NICK

interface Nick {
  user: User;
  nick: string;
}

export const NICK = "MESSAGE/INCOMING/NICK";

export type NickAction = IncomingMessageAction<typeof NICK, Nick>;

export const receiveNick: IncomingMessageActionCreator<NickAction, User> = (
  serverKey,
  user,
  params,
) => ({
  type: NICK,
  payload: { user, nick: params[0] },
  route: { serverKey, channelKey: STATUS },
});

// NOTICE

interface ServerNotice {
  server: Server;
  text: string;
}

interface ChannelNotice {
  user: User;
  channel: string;
  text: string;
}

interface UserNotice {
  user: User;
  text: string;
}

export const SERVER_NOTICE = "MESSAGE/INCOMING/SERVER_NOTICE";

export const NOTICE_CHANNEL = "MESSAGE/INCOMING/CHANNEL_NOTICE";

export const USER_NOTICE = "MESSAGE/INCOMING/USER_NOTICE";

export type ServerNoticeAction = IncomingMessageAction<
  typeof SERVER_NOTICE,
  ServerNotice
>;

export type ChannelNoticeAction = IncomingMessageAction<
  typeof NOTICE_CHANNEL,
  ChannelNotice
>;

export type UserNoticeAction = IncomingMessageAction<
  typeof USER_NOTICE,
  UserNotice
>;

export const receiveNotice: IncomingMessageActionCreator<
  ServerNoticeAction | ChannelNoticeAction | UserNoticeAction,
  Prefix
> = (serverKey, prefix, params) => {
  if (isPrefixServer(prefix)) {
    return {
      type: SERVER_NOTICE,
      payload: { server: prefix as Server, text: params[1] },
      route: { serverKey, channelKey: STATUS },
    };
  }

  return isChannel(params[0])
    ? {
        type: NOTICE_CHANNEL,
        payload: { user: prefix as User, channel: params[0], text: params[1] },
        route: { serverKey, channelKey: params[0] },
      }
    : {
        type: USER_NOTICE,
        payload: { user: prefix as User, text: params[1] },
        route: { serverKey, channelKey: STATUS },
      };
};

// PART

interface Part {
  user: User;
  channel: string;
  message?: string;
}

export const PART = "MESSAGE/INCOMING/PART";

export type PartAction = IncomingMessageAction<typeof PART, Part>;

export const receivePart: IncomingMessageActionCreator<PartAction, User> = (
  serverKey,
  user,
  params,
) => ({
  type: PART,
  payload: { user, channel: params[0], message: params[1] },
  route: { serverKey, channelKey: params[0] },
});

// PING

interface ServerPing {
  key: string;
}

export const SERVER_PING = "MESSAGE/INCOMING/SERVER_PING";

export type ServerPingAction = IncomingMessageAction<
  typeof SERVER_PING,
  ServerPing
>;

export const receivePing: IncomingMessageActionCreator<ServerPingAction> = (
  serverKey,
  _,
  params,
) => ({
  type: SERVER_PING,
  payload: { key: params.join(" ") },
  route: { serverKey, channelKey: STATUS },
});

// PRIVMSG

interface Privmsg {
  user: User;
  text: string;
}

export const PRIVMSG = "MESSAGE/INCOMING/PRIVMSG";

export type PrivmsgAction = IncomingMessageAction<typeof PRIVMSG, Privmsg>;

export const receivePrivmsg: IncomingMessageActionCreator<
  PrivmsgAction,
  User
> = (serverKey, user, params) => {
  const target = params[0];

  return {
    type: PRIVMSG,
    payload: { user, text: params[1] },
    route: { serverKey, channelKey: isChannel(target) ? target : user.nick },
  };
};

// 004 RPL_MYINFO

interface ReplyMyInfo {
  serverName: string;
  version: string;
  availableUserModes: string[];
  availableChannelModes: string[];
}

export const RPL_MYINFO = "MESSAGE/INCOMING/RPL_MYINFO";

export type ReplyMyInfoAction = IncomingMessageAction<
  typeof RPL_MYINFO,
  ReplyMyInfo
>;

export const receiveReplyMyInfo: IncomingMessageActionCreator<
  ReplyMyInfoAction,
  Server
> = (serverKey, _, params) => {
  const [, serverName, version] = params;
  const availableUserModes = params[3].split("");
  const availableChannelModes = params[4].split("");

  return {
    type: RPL_MYINFO,
    payload: {
      serverName,
      version,
      availableUserModes,
      availableChannelModes,
    },
    route: { serverKey, channelKey: STATUS },
  };
};
