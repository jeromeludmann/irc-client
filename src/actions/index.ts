export const SET_ACTIVE_WINDOW = "SET_ACTIVE_WINDOW";

export interface SetActiveWindow {
  type: typeof SET_ACTIVE_WINDOW;
  payload: { server: string; channel: string };
}

export function setActiveWindow(server: string, channel: string) {
  return {
    type: SET_ACTIVE_WINDOW,
    payload: { server, channel },
  };
}
