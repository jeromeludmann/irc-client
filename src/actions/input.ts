import { ChannelScope } from "@app/types";

export const InputActionTypes = {
  CHANGED: "INPUT/CHANGED",
  SENT: "INPUT/SENT",
};

export interface InputChangedAction {
  type: typeof InputActionTypes.CHANGED;
  scope: ChannelScope;
  payload: { value: string };
}

export interface InputSentAction {
  type: typeof InputActionTypes.SENT;
  scope: ChannelScope;
  payload: { value: string };
}

export function changeInput(scope: ChannelScope, input: string) {
  return {
    type: InputActionTypes.CHANGED,
    scope,
    payload: { value: input },
  };
}

export function sendInput(scope: ChannelScope, input: string) {
  return {
    type: InputActionTypes.SENT,
    scope,
    payload: { value: input },
  };
}
