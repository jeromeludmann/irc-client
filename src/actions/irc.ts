import { ClientMessage, Command } from "@app/irc/Message";

export const SEND_COMMAND = "SEND_COMMAND";

export interface SendCommand {
  type: typeof SEND_COMMAND;
  payload: ClientMessage;
}

export const whois = (nick: string) => {
  return command("whois", nick);
};

export const join = (channel: string) => {
  return command("join", channel);
};

export const privmsg = (target: string, text: string) => {
  return command("privmsg", target, text);
};

function command(name: Command, ...parameters: string[]): SendCommand {
  return {
    type: SEND_COMMAND,
    payload: { command: name, parameters },
  };
}
