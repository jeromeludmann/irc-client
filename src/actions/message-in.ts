import { Action } from "redux";
import { Route, STATUS_BUFFER } from "@app/Route";

export interface IncomingMessageAction<T, M> extends Action<T> {
  payload: M;
  route: Route;
}

export type IncomingMessageActionCreator<A, P = Prefix> = (
  serverKey: string,
  prefix: P,
  params: string[],
) => A;

export type Prefix = Server | User;

export type Server = string;

export type User = {
  nick: string;
  user: string;
  host: string;
};

export const isChannel = (channel: string) => /^(&|#|\+|!)/.test(channel);

export const isPrefixServer = (prefix: Prefix) => typeof prefix === "string";

export const isPrefixUser = (prefix: Prefix) => !isPrefixServer(prefix);

// JOIN

interface Join {
  user: User;
  channel: string;
}

export const JOIN = "MESSAGE/INCOMING/JOIN";

export type JoinAction = IncomingMessageAction<typeof JOIN, Join>;

export const join: IncomingMessageActionCreator<JoinAction, User> = (
  serverKey,
  user,
  params,
) => ({
  type: JOIN,
  payload: { user, channel: params[0] },
  route: { serverKey, bufferKey: params[0] },
});

// NICK

interface Nick {
  user: User;
  nick: string;
}

export const NICK = "MESSAGE/INCOMING/NICK";

export type NickAction = IncomingMessageAction<typeof NICK, Nick>;

export const nick: IncomingMessageActionCreator<NickAction, User> = (
  serverKey,
  user,
  params,
) => ({
  type: NICK,
  payload: { user, nick: params[0] },
  route: { serverKey, bufferKey: STATUS_BUFFER },
});

// NOTICE

interface ServerNotice {
  server: Server;
  text: string;
}

interface ChannelNotice {
  user: User;
  channel: string;
  text: string;
}

interface UserNotice {
  user: User;
  text: string;
}

export const SERVER_NOTICE = "MESSAGE/INCOMING/SERVER_NOTICE";

export const CHANNEL_NOTICE = "MESSAGE/INCOMING/CHANNEL_NOTICE";

export const USER_NOTICE = "MESSAGE/INCOMING/USER_NOTICE";

export type ServerNoticeAction = IncomingMessageAction<
  typeof SERVER_NOTICE,
  ServerNotice
>;

export type ChannelNoticeAction = IncomingMessageAction<
  typeof CHANNEL_NOTICE,
  ChannelNotice
>;

export type UserNoticeAction = IncomingMessageAction<
  typeof USER_NOTICE,
  UserNotice
>;

export const notice: IncomingMessageActionCreator<
  ServerNoticeAction | ChannelNoticeAction | UserNoticeAction
> = (serverKey, prefix, params) => {
  if (isPrefixServer(prefix)) {
    return {
      type: SERVER_NOTICE,
      payload: { server: prefix as Server, text: params[1] },
      route: { serverKey, bufferKey: STATUS_BUFFER },
    };
  }

  return isChannel(params[0])
    ? {
        type: CHANNEL_NOTICE,
        payload: { user: prefix as User, channel: params[0], text: params[1] },
        route: { serverKey, bufferKey: params[0] },
      }
    : {
        type: USER_NOTICE,
        payload: { user: prefix as User, text: params[1] },
        route: { serverKey, bufferKey: STATUS_BUFFER },
      };
};

// PING

interface ServerPing {
  key: string;
}

export const SERVER_PING = "MESSAGE/INCOMING/SERVER_PING";

export type ServerPingAction = IncomingMessageAction<
  typeof SERVER_PING,
  ServerPing
>;

export const ping: IncomingMessageActionCreator<ServerPingAction, Server> = (
  serverKey,
  _server,
  params,
) => ({
  type: SERVER_PING,
  payload: { key: params.join(" ") },
  route: { serverKey, bufferKey: STATUS_BUFFER },
});

// PRIVMSG

interface ChannelPrivmsg {
  user: User;
  channel: string;
  text: string;
}

interface UserPrivmsg {
  user: User;
  text: string;
}

export const CHANNEL_PRIVMSG = "MESSAGE/INCOMING/CHANNEL_PRIVMSG";

export const USER_PRIVMSG = "MESSAGE/INCOMING/USER_PRIVMSG";

export type ChannelPrivmsgAction = IncomingMessageAction<
  typeof CHANNEL_PRIVMSG,
  ChannelPrivmsg
>;

export type UserPrivmsgAction = IncomingMessageAction<
  typeof USER_PRIVMSG,
  UserPrivmsg
>;

export const privmsg: IncomingMessageActionCreator<
  ChannelPrivmsgAction | UserPrivmsgAction,
  User
> = (serverKey, user, params) => {
  const target = params[0];
  const text = params[1];

  return isChannel(target)
    ? {
        type: CHANNEL_PRIVMSG,
        payload: { user, channel: target, text },
        route: { serverKey, bufferKey: target },
      }
    : {
        type: USER_PRIVMSG,
        payload: { user, text },
        route: { serverKey, bufferKey: user.nick },
      };
};
