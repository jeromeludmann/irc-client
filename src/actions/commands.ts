import { Action } from "redux";
import {
  connectServer,
  disconnectServer,
  sendMessage,
  sendRaw,
  DisconnectServerAction,
  ConnectServerAction,
  SendRawMessageAction,
} from "@app/actions/socket";
import { closeWindow } from "@app/actions/ui";
import { Route, RoutedAction, STATUS } from "@app/Route";

export type Command<A = Action<string>> = {
  description: string;
  syntax: string;
  regexp: RegExp;
  callback: (route: Route, ...args: string[]) => A;
};

type CommandRegistry = { [command: string]: Command };

// close

export const close: Command = {
  description: "Closes the current window",
  syntax: "",
  regexp: /\s*/,
  callback: route => closeWindow(route),
};

// connect

export const connect: Command<ConnectServerAction> = {
  description: "Connects to a server",
  syntax: "[-n] <host> [port]",
  regexp: /^(?:(-n)\s+)?(\S+)(?:\s+(\d{1,5})?)?$/,
  callback: (route, option, host, port?) =>
    connectServer(
      route.serverKey,
      host,
      port ? +port : undefined,
      option === "-n",
    ),
};

// disconnect

export const disconnect: Command<DisconnectServerAction> = {
  description: "Disconnects from the current server",
  syntax: "[quit message]",
  regexp: /^(.*)$/,
  callback: (route, quitMessage?) =>
    disconnectServer(route.serverKey, quitMessage),
};

// help

export const PRINT_HELP_BY_DEFAULT = "COMMAND/PRINT_HELP_BY_DEFAULT";
export const PRINT_HELP_ABOUT_COMMAND = "COMMAND/PRINT_HELP_ABOUT_COMMAND";

export interface PrintHelpByDefaultAction
  extends RoutedAction<typeof PRINT_HELP_BY_DEFAULT> {
  payload: { commands: CommandRegistry };
}

export interface PrintHelpAboutCommandAction
  extends RoutedAction<typeof PRINT_HELP_ABOUT_COMMAND> {
  payload: { command: Command & { name: string } };
}

export const help: Command<
  PrintHelpByDefaultAction | PrintHelpAboutCommandAction
> = {
  description: "Prints help about available commands",
  syntax: "[command]",
  regexp: /^(\S+)?$/,
  callback: (route, commandName?) => {
    if (commandName) {
      commandName = commandName.toLowerCase();
    }

    if (commandName && registry.hasOwnProperty(commandName)) {
      return {
        type: PRINT_HELP_ABOUT_COMMAND,
        payload: { command: { ...registry[commandName], name: commandName } },
        route: { ...route, channelKey: STATUS },
      };
    }

    return {
      type: PRINT_HELP_BY_DEFAULT,
      payload: { commands: registry },
      route: { ...route, channelKey: STATUS },
    };
  },
};

// join

export const join: Command<SendRawMessageAction> = {
  description: "Joins a channel",
  syntax: "<channel>",
  regexp: /^([&|#|\+|!]\S+)$/, // TODO extract (see Route.ts)
  callback: (route, channel) => sendMessage(route.serverKey, "join", channel),
};

// msg

export const msg: Command<SendRawMessageAction> = {
  description: "Sends a message to a channel or a nick",
  syntax: "<channel or nick> <text>",
  regexp: /^(\S+)\s+(.+)$/,
  callback: (route, channel, text) =>
    sendMessage(route.serverKey, "privmsg", channel, text),
};

// part

export const part: Command<SendRawMessageAction> = {
  description: "Leaves the channel",
  syntax: "<channel> [text]",
  regexp: /^(\S+)(?:\s+(.+))?$/,
  callback: (route, channel, text) =>
    sendMessage(route.serverKey, "part", channel, text),
};

// quit

export const quit: Command<SendRawMessageAction> = {
  description: "Disconnects from the current server",
  syntax: "[quit message]",
  regexp: /^(.*)$/,
  callback: (route, text) => sendMessage(route.serverKey, "quit", text),
};

// raw (quote)

export const raw: Command<SendRawMessageAction> = {
  description: "Sends a raw message",
  syntax: "<raw message>",
  regexp: /^(.+)$/,
  callback: (route, message) => sendRaw(route.serverKey, message),
};

export const registry: CommandRegistry = {
  close,
  connect,
  disconnect,
  help,
  join,
  msg,
  part,
  quit,
  quote: raw,
  raw,
};
