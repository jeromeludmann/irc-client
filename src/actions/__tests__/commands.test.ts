import { commands } from "@app/actions/commands";

describe("commands actions", () => {
  const route = { serverKey: "server1", channelKey: "#channel" };

  it("should be able to call /close", () => {
    expect(commands.close.callback(route)).toMatchSnapshot();
  });

  it("should be able to call /join", () => {
    expect(commands.join.callback(route, "#channel")).toMatchSnapshot();
  });

  it("should be able to call /msg", () => {
    expect(commands.msg.callback(route, "#channel", "hello")).toMatchSnapshot();
  });

  it("should be able to call /nick", () => {
    expect(commands.nick.callback(route, "new_nick")).toMatchSnapshot();
  });

  it("should be able to call /part", () => {
    expect(
      commands.part.callback(route, "#channel", "Goodbye!"),
    ).toMatchSnapshot();
  });

  it("should be able to call /ping", () => {
    expect(commands.ping.callback(route)).toMatchSnapshot();
  });

  describe("should be able to call /server", () => {
    it("without option", () => {
      expect(
        commands.server.callback(route, "-n", "irc.network", "6667"),
      ).toMatchSnapshot();
    });

    it("with option -n", () => {
      expect(
        commands.server.callback(route, "irc.network", "6667"),
      ).toMatchSnapshot();
    });
  });

  it("should be able to call /quit", () => {
    expect(commands.quit.callback(route, "Goodbye!")).toMatchSnapshot();
  });

  it("should be able to call /raw", () => {
    expect(
      commands.raw.callback(route, "PRIVMSG #channel :hello world"),
    ).toMatchSnapshot();
  });
});
