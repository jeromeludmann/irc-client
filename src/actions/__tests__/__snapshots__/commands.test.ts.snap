// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands actions should be able to call /close 1`] = `
Object {
  "route": Object {
    "channelKey": "#channel",
    "serverKey": "server1",
  },
  "type": "UI/CLOSE_WINDOW",
}
`;

exports[`commands actions should be able to call /join 1`] = `
Object {
  "payload": Object {
    "raw": "JOIN #channel",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`commands actions should be able to call /msg 1`] = `
Object {
  "embeddedAction": Object {
    "payload": Object {
      "text": "hello",
    },
    "route": Object {
      "bufferKey": "#channel",
      "serverKey": "server1",
    },
    "type": "MESSAGE/SEND_PRIVMSG",
  },
  "payload": Object {
    "raw": "PRIVMSG #channel hello",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`commands actions should be able to call /nick 1`] = `
Object {
  "payload": Object {
    "raw": "NICK new_nick",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`commands actions should be able to call /part 1`] = `
Object {
  "payload": Object {
    "raw": "PART #channel Goodbye!",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`commands actions should be able to call /ping 1`] = `
Object {
  "embeddedAction": Object {
    "payload": Object {
      "key": "1536211245164",
    },
    "route": Object {
      "bufferKey": "@status",
      "serverKey": "server1",
    },
    "type": "MESSAGE/SEND_PING_TO_SERVER",
  },
  "payload": Object {
    "raw": "PING 1536211245164",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`commands actions should be able to call /quit 1`] = `
Object {
  "payload": Object {
    "quitMessage": "Goodbye!",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/DISCONNECT_FROM_SERVER",
}
`;

exports[`commands actions should be able to call /raw 1`] = `
Object {
  "payload": Object {
    "raw": "PRIVMSG #channel :hello world",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`commands actions should be able to call /server with option -n 1`] = `
Object {
  "payload": Object {
    "host": "6667",
    "newConnection": false,
    "port": 6667,
  },
  "route": Object {
    "bufferKey": "@status",
    "serverKey": "server1",
  },
  "type": "SOCKET/CONNECT_TO_SERVER",
}
`;

exports[`commands actions should be able to call /server without option 1`] = `
Object {
  "payload": Object {
    "host": "irc.network",
    "newConnection": true,
    "port": 6667,
  },
  "route": Object {
    "bufferKey": "@status",
    "serverKey": "server1",
  },
  "type": "SOCKET/CONNECT_TO_SERVER",
}
`;
